relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -733,48 +695,119 @@ mod tests {
             &mut metrics,
         );
 
-        assert_eq!(metrics.len(), 5, "{:?}", metrics);
-
-        assert_eq!(metrics[0].name, "d:transactions/measurements.foo@none");
-        assert_eq!(
-            metrics[1].name,
-            "d:transactions/measurements.lcp@millisecond"
-        );
-        assert_eq!(
-            metrics[2].name,
-            "d:transactions/breakdowns.span_ops.ops.react.mount@millisecond"
-        );
-
-        let duration_metric = &metrics[3];
-        assert_eq!(duration_metric.name, "d:transactions/duration@millisecond");
-        if let MetricValue::Distribution(value) = duration_metric.value {
-            assert_eq!(value, 59000.0);
-        } else {
-            panic!(); // Duration must be set
-        }
-
-        let user_metric = &metrics[4];
-        assert_eq!(user_metric.name, "s:transactions/user@none");
-        assert!(matches!(user_metric.value, MetricValue::Set(_)));
-
-        assert_eq!(metrics[1].tags["measurement_rating"], "meh");
-
-        for metric in &metrics[0..4] {
-            assert!(matches!(metric.value, MetricValue::Distribution(_)));
-        }
-
-        for metric in metrics {
-            assert_eq!(metric.tags["release"], "1.2.3");
-            assert_eq!(metric.tags["dist"], "foo");
-            assert_eq!(metric.tags["environment"], "fake_environment");
-            assert_eq!(metric.tags["transaction"], "mytransaction");
-            assert_eq!(metric.tags["fOO"], "bar");
-            assert_eq!(metric.tags["http.method"], "POST");
-            assert_eq!(metric.tags["transaction.status"], "ok");
-            assert_eq!(metric.tags["transaction.op"], "myop");
-            assert_eq!(metric.tags["platform"], "javascript");
-            assert!(!metric.tags.contains_key("bogus"));
-        }
+        insta::assert_debug_snapshot!(metrics, @r###"
```

Comment:
The test diff is large because a few tests used the metrics allowlist to filter down metrics returned. In the process, I decided to replace ad-hoc assert statements with `insta::assert_debug_snapshot!` where possible.

Reply:
> The test diff is large because a few tests used the metrics allowlist to filter down metrics returned.

Definitely out of the scope of this PR, but tests should not interfere with other tests.

Reply:
I like using snapshots (vs ad-hoc asserts) a lot. However, making this change in this PR makes focusing on the relevant aspects harder, vs having another PR for the refactor. No action required on this PR, just a note for future PRs.